File: app/src/main/java/com/example/shop/data/remote/dto/CartDtos.kt
package com.example.shop.data.remote.dto

import com.example.shop.data.model.CartItem

data class CartResponse(
    val items: List<CartItemResponse>,
    val totalPrice: Double
)

data class CartItemResponse(
    val id: Int,
    val productId: Int,
    val productName: String,
    val price: Double,
    val imageUrl: String,
    val quantity: Int,
    val lineTotal: Double
)

data class AddCartItemRequest(
    val productId: Int,
    val quantity: Int
)

data class UpdateCartItemRequest(
    val quantity: Int
)

fun CartItemResponse.toCartItem(): CartItem {
    return CartItem(
        id = id,
        productId = productId.toString(),
        productName = productName,
        quantity = quantity,
        price = price,
        imageUrl = imageUrl
    )
}

File: backend/ShopApi/Dtos/CartDtos.cs
using System.ComponentModel.DataAnnotations;

namespace ShopApi.Dtos;

public record CartResponse(
    List<CartItemResponse> Items,
    decimal TotalPrice);

public record CartItemResponse(
    int Id,
    int ProductId,
    string ProductName,
    decimal Price,
    string ImageUrl,
    int Quantity,
    decimal LineTotal);

public record AddCartItemRequest(
    int ProductId,
    [Range(1, int.MaxValue)] int Quantity);

public record UpdateCartItemRequest(
    [Range(1, int.MaxValue)] int Quantity);
